File: part2_oop/src/Advertiser.cpp
#include "Advertiser.hpp"

const std::string help_msg = "Advertiser Class Fields :\n"
                             "id : unique number assigned to each advertiser.\n"
                             "name : name of the advertiser.\n"
                             "clicks : number of clicks on the advertiser ads.\n"
                             "views : number of views of the advertiser ads.\n";
const std::string describeMe_msg = "Advertiser Class methods :\n"
                                   "getName() : gives name of the advertiser.\n"
                                   "setName(new_name) : changes name of the advetiser to new_name.\n"
                                   "help() : gives information about this class fields.\n"
                                   "getClicks() : gives number of clicks on the advertiser ads.\n"
                                   "getViews() : gives number of views of the advertiser ads.\n"
                                   "incClicks() : increments number of clicks by one.\n"
                                   "incViews() : increments number of views by one.\n"
                                   "describeMe() : describes the methods action.\n";
Advertiser::Advertiser(int _id, std::string _name) 
    : BaseAdvertising(_id), name(_name) {}

std::string Advertiser::getName() {
    return name;
}
void Advertiser::setName(std::string _name) {
    name = _name;
}
std::string Advertiser::help() {
    return help_msg;
}
int Advertiser::total_clicks = 0;
int Advertiser::getTotalClicks() {
    return total_clicks;
}
std::string Advertiser::describeMe() {
    return describeMe_msg;
}
void Advertiser::incClicks() {
    clicks++;
    total_clicks++;
}

File: part3_python/advertiser.py
from base_advertising import BaseAdvertising

class Advertiser(BaseAdvertising):
    total_clicks = 0

    def __init__(self, _id, _name):
        super().__init__(_id)
        self.name = _name

    def get_name(self):
        return self.name

    def set_name(self, _name):
        self.name = _name

    @staticmethod
    def help():
        return "Advertiser Class Fields:\n" \
               "id: unique number assigned to each advertiser.\n" \
               "name: name of the advertiser.\n" \
               "clicks: number of clicks on the advertiser ads.\n" \
               "views: number of views of the advertiser ads.\n"

    @staticmethod
    def get_total_clicks():
        return Advertiser.total_clicks

    def describe_me(self):
        return "Advertiser Class methods:\n" \
               "get_name(): gives the name of the advertiser.\n" \
               "set_name(new_name): changes the name of the advertiser to new_name.\n" \
               "help(): gives information about this class fields.\n" \
               "get_clicks(): gives the number of clicks on the advertiser ads.\n" \
               "get_views(): gives the number of views of the advertiser ads.\n" \
               "inc_clicks(): increments the number of clicks by one.\n" \
               "inc_views(): increments the number of views by one.\n" \
               "describe_me(): describes the methods action.\n"

    def inc_clicks(self):
        self.clicks += 1
        Advertiser.total_clicks += 1
